src/components/career-form/career-file-upload-type.tsx

```diff
@@ -29,40 +40,159 @@ export const CareerFileUploadType = ({
 }: CareerFileUploadTypeProps) => {
   const fileName = file.name.split('.');
   const name = `category_select.${fileName[0]}`;
-  const state = watch(name);
+  const state = watch(name, 'default');
+
+  useEffect(() => {
+    register(name, {
+      validate: (value) => {
+        return Boolean(value) && value !== 'default';
+      },
+    });
+    setValue(name, 'default', {
+      shouldDirty: true,
+    });
+  }, []);
 
   const onChange = (event) => {
-    setValue(name, event.target.value);
+    setValue(name, event, {
+      shouldDirty: true,
+    });
     clearError(name);
   };
 
   return (
-    <div>
-      <StyledSelect
+    <SelectWrapper>
+      <StyledListboxInput
         id={name}
-        {...register(name, {
-          validate: (value) => value !== 'default',
-        })}
+        aria-labelledby={name}
         value={state || 'default'}
         onChange={onChange}
       >
-        <option value={'default'} disabled>
-          File category
-        </option>
-        <option value="cv">CV</option>
-        <option value="cover-letter">Cover-Letter</option>
-        <option value="other">other</option>
-      </StyledSelect>
+        <ListboxButton arrow={<SelectArrow />} />
+        <ListboxPopover portal={false}>
+          <ListboxList>
+            <ListboxOption value={'default'} disabled>
+              File category
+            </ListboxOption>
+            <ListboxOption value="cv">CV</ListboxOption>
+            <ListboxOption value="cover-letter">Cover-Letter</ListboxOption>
+            <ListboxOption value="other">other</ListboxOption>
+          </ListboxList>
+        </ListboxPopover>
+      </StyledListboxInput>
       {errors.category_select?.[fileName[0]] && (
-        <FormError
-          error={{ message: 'Deine Auswahl fehlt' }}
-          lineHeight={100}
-        />
+        <FormErrorWrapper>
+          <FormError
+            error={{ message: 'Deine Auswahl fehlt' }}
+            lineHeight={90}
+          />
+        </FormErrorWrapper>
       )}
-    </div>
+    </SelectWrapper>
   );
 };
 
-const StyledSelect = styled.select`
-  border-radius: 4px;
+const SelectWrapper = styled.div`
+  display: block;
+  position: relative;
+
+  background: none;
+  border: none;
+  width: 100%;
+  margin-top: 10px;
+
+  ${up('md')} {
+    width: 144px;
+    margin-top: 0px;
+  }
+`;
+
+const FormErrorWrapper = styled.div`
+  margin-left: 16px;
+`;
+
+const StyledListboxInput = styled(ListboxInput)`
+  position: relative;
+  background: ${rgba('#ffffff', 0.1)};
+  border-radius: 20px;
+  border: none;
+  color: white;
+  font-weight: bold;
+  font-size: 14px;
+  cursor: pointer;  
+  width: calc(100% + 25px);
+  line-height: 110%;
+  
+  &:hover{
+    background: ${rgba('#ffffff', 0.2)};
+  }
+  
+  ${up('md')} {
+    width: 144px;   
+    height: 32px;
+  }
+  
+  [data-reach-listbox-button] {
```

Comment:
I guess that's the recommended way to style reach ui ? I didn't expect that they support approaches like styled component although it would have been nice here instead of squeezing in so many rules. 

It's quite some css, do we have any recommendations when and how to split css specific componnets into separate files @feedm3 ?

Reply:
Yep, that CSS is a lot. But actually I have no idea right now on how to split the CSS. It looks like it's necessary to style the reach-ui components. If thats recommended by reach-ui it would be ok for me.